src/mdesktopaction.h

```diff
@@ -0,0 +1,43 @@
+/***************************************************************************
+** This file was derived from the MDesktopAction implementation in the
+** libmeegotouch library.
+**
+** Copyright (C) 2021 Chupligin Sergey (NeoChapay) <neochapay@gmail.com>
+**
+** This library is free software; you can redistribute it and/or
+** modify it under the terms of the GNU Lesser General Public
+** License version 2.1 as published by the Free Software Foundation
+** and appearing in the file LICENSE.LGPL included in the packaging
+** of this file.
+**
+****************************************************************************/
+
+#ifndef MDESKTOPACTION_H
+#define MDESKTOPACTION_H
+
+#include <mlite-global.h>
+#include <QString>
+
+class MDesktopEntryPrivate;
+
+class MLITESHARED_EXPORT MDesktopAction
+{
+
+public:
+    MDesktopAction(const QString &fileName, const QString &action);
+    virtual ~MDesktopAction();
+
+    QString name() const;
+    QString nameUnlocalized() const;
+    QString getExec() const;
+    QString icon() const;
+    QString actionName() const;
+    QString getActionEntryName() const;
+
+protected:
+    /*! \internal */
+    //! Pointer to the private class
+    MDesktopEntryPrivate *const d_ptr;
```

Comment:
Shouldn't private part of MDesktop**Action** be called MDesktop**Action**Private rather than MDesktop**Entry**Private?

Reply:
Action is part of entry. I think it's overcoding create separate class for one variable

Reply:
It doesn't need to be made too complicated, e.g. no need for separate file. Could just declare a private class in the .cpp and inherit the existing one.

Then again I'm not sure about that inheritance at all. It's rereading the desktop file separately for each action when the actions really only need a few properties available.